autoscale/src/main/java/com/sequenceiq/periscope/service/ScalingActivityService.java

```diff
@@ -66,10 +67,28 @@ public List<Long> findAllIdsOlderThanDurationForCluster(Cluster cluster, long du
                 .stream().map(ScalingActivity::getId).collect(Collectors.toList());
     }
 
+    public List<ScalingActivity> findAllInGivenIntervalForCluster(Cluster cluster, long durationInMinutes, Pageable pageable) {
+        Date startTimeAfter = new Date(Instant.now().minus(durationInMinutes, ChronoUnit.MINUTES).toEpochMilli());
+        return scalingActivityRepository.findAllByClusterWithStartTimeAfter(cluster.getId(), startTimeAfter, pageable);
+    }
+
+    public List<ScalingActivity> findAllWithTriggerFailed(Cluster cluster, long durationInMinutes, Pageable pageable) {
```

Comment:
Can this method name be changed to `findAllByFailedStatuses` or something similar?

Reply:
done